### PR TITLE
[Module Builder] - Preview Screen

### PR DESCRIPTION
**Pull request by christineNg1028**
## Notion ticket link

[Module Builder - Preview Screen](https://www.notion.so/uwblueprintexecs/Preview-Screen-9c29f26880e84f2d94f9edaa9b60deeb)

## Implementation description
* Preview screen for module builder, mobile and desktop versions
* Props to determine buttons, button text, and progress, takes all preview content as children
* Print/save buttons disappear when false, prev/next buttons disable when false
* Modified progress bar styling

## Steps to test
1. Ensure consistent with mockups and that all functionality is included

**Usage**
Screenshot 2021-08-10 121355
Set props and pass in any content as children

**Desktop**
Screenshot 2021-08-10 121429
Note: must scroll to see buttons when content overflows

**Mobile**
Screenshot 2021-08-10 121513
Note: buttons fixed to bottom when content doesn't overflow

## What should reviewers focus on?
* Styling may need updates

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by nimwijetunga — COMMENTED** [review]
Mostly looks good! However, for some reason the progress bar doesn't display for me (on Mac). This is how i render the component:
```
<ModulePreview previous={true} next={true} save={false} print={true} variant={"mobile"} progressValue={40}>
                <Button>test</Button>
                <Button>test2</Button>
                <Button>test3</Button>
            </ModulePreview>
```

File: src/components/modulePreview/index.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
+import React from "react";
+import { Box, Flex, Progress } from "@chakra-ui/react";
+import { Button } from "..";
```
Comment (tawnyzhao) [inline src/components/modulePreview/index.tsx:3]: This import is wrong, should be from chakra-ui

File: src/definitions/chakra/theme/components/button.ts [hunk 2]
```diff
@@ -21,6 +21,18 @@ const Button = {
...
+            color: "black",
+            borderRadius: 0,
+            textTransform: "uppercase",
+        },
+        moduleBlack: {
+            bg: "black",
+            color: "white",
+            borderRadius: 0,
+            textTransform: "uppercase",
+        }
```
Comment (tawnyzhao) [inline src/definitions/chakra/theme/components/button.ts:35]: lint: add a comma

**Review by tawnyzhao — CHANGES_REQUESTED** [review]

**Review by tawnyzhao — APPROVED** [review]
👍🏻

**Review by chamod-gamage — APPROVED** [review]
LGTM! Great work~

**Pull request merged**